### PR TITLE
Add support for custom configuration

### PR DESCRIPTION
**Pull request by Spellchaser**
If someone has a cluster that isn't easily configured with InClusterConfig, this allows them to use a custom config. This does not change the user experience for normal users.

### CODE REVIEW

**Comment by asobti** [comment]
Could you resolve the conflicts with this PR

File: kubernetes/kubernetes.go [hunk 1]
```diff
@@ -7,20 +7,57 @@ import (
...
+                config := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
+                tmpKubeConfig, err := config.ClientConfig()
+                if err != nil {
+                        return nil, fmt.Errorf("Failed to load local kube config: %v", err)
+                }
+                kubeConfig = *tmpKubeConfig;
+        }
+
+	if apiserverHost, override := cfg.ClusterAPIServerHost(); override {
+		fmt.Printf("API server host overriden to: %s\n", apiserverHost)
```
Comment (asobti) [inline kubernetes/kubernetes.go]: This file is still using `fmt` (pre-glog commit) but is no longer importing the fmt package.

Here are the errors I'm seeing:

```
GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -o kube-monkey
# github.com/asobti/kube-monkey/kubernetes
kubernetes/kubernetes.go:16:22: undefined: os
kubernetes/kubernetes.go:31:37: undefined: fmt
kubernetes/kubernetes.go:41:37: undefined: fmt
kubernetes/kubernetes.go:47:3: undefined: fmt
kubernetes/kubernetes.go:48:3: undefined: config
```

Comment (Spellchaser) [reply kubernetes/kubernetes.go]: Ahh yes, I was on the go and unable to integrate the glog changes. It should now be using it glog style.

**Comment by Spellchaser** [comment]
I developed this branch separately from glog

Resolved the conflicts correctly and verified the build locally :) ready for review

**Pull request merged**